augur/frequencies.py

```diff
@@ -108,16 +108,16 @@ def run(args):
     stiffness = args.stiffness
     inertia = args.inertia
 
-    if args.method == "kde":
+    # For KDE
+    weights = None
+    weights_attribute = None
+
+    if args.method == "kde" and args.weights:
```

Comment:
_non-blocking_

I don't see anything enforcing that `--weights` and `--weights-attribute` must be provided together...Would there be a downstream error if `weights_attribute` is still `None`?

Reply:
Yes, I just checked with this Cram test:

```
  $ ${AUGUR} frequencies \
  >  --method kde \
  >  --tree tree.nwk \
  >  --metadata metadata.tsv \
  >  --weights weights.json \
  >  --output tip-frequencies.json >/dev/null
  Traceback (most recent call last):
    File "…/augur/__init__.py", line 66, in run
      return args.__command__.run(args)
    File "…/augur/frequencies.py", line 185, in run
      frequencies = kde_frequencies.estimate(tree)
    File "…/augur/frequency_estimators.py", line 1158, in estimate
      weights_represented = set([
    File "…/augur/frequency_estimators.py", line 1159, in <listcomp>
      tip.attr[self.weights_attribute]
  KeyError: None
```

This has been around for as long as the argument was available. I'll make a new issue to add better argument validation to frequencies.

Reply:
#1414